guvnor-ala/guvnor-ala-wildfly/guvnor-ala-wildfly-ui/guvnor-ala-wildfly-ui-client/src/main/java/org/guvnor/ala/ui/wildfly/client/provider/WF10ProviderConfigView.java

```diff
@@ -0,0 +1,353 @@
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.ForEvent;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+import static org.guvnor.ala.ui.client.widget.StyleHelper.setFormStatus;
+import static org.guvnor.ala.ui.wildfly.client.resources.i18n.GuvnorAlaWildflyUIConstants.WF10ProviderConfigView_AllParamsNeedsCompletionForValidationMessage;
+import static org.guvnor.ala.ui.wildfly.client.resources.i18n.GuvnorAlaWildflyUIConstants.WF10ProviderConfigView_TestConnectionFailMessage;
+import static org.guvnor.ala.ui.wildfly.client.resources.i18n.GuvnorAlaWildflyUIConstants.WF10ProviderConfigView_TestConnectionSuccessfulMessage;
+import static org.guvnor.ala.ui.wildfly.client.resources.i18n.GuvnorAlaWildflyUIConstants.WF10ProviderConfigView_TestConnectionUnExpectedErrorMessage;
+import static org.uberfire.commons.validation.PortablePreconditions.checkNotNull;
+
+@Dependent
+@Templated
+public class WF10ProviderConfigView
+        implements IsElement,
+                   WF10ProviderConfigPresenter.View {
+
+    @Inject
+    @DataField("provider-name-form")
+    private Div providerNameForm;
+
+    @DataField("provider-type-name")
+    private HTMLElement providerTypeName = Window.getDocument().createElement("strong");
+
+    @Inject
+    @DataField("provider-name")
+    private TextInput name;
+
+    @Inject
+    @DataField("host-form")
+    private Div hostForm;
+
+    @Inject
+    @DataField("host")
+    private TextInput host;
+
+    @Inject
+    @DataField("port-form")
+    private Div portForm;
+
+    @Inject
+    @DataField("port")
+    private TextInput port;
+
+    @Inject
+    @DataField("management-port")
+    private TextInput managementPort;
+
+    @Inject
+    @DataField("management-port-form")
+    private Div managementPortForm;
+
+    @Inject
+    @DataField("username-form")
+    private Div usernameForm;
+
+    @Inject
+    @DataField("username")
+    private TextInput username;
+
+    @Inject
+    @DataField("password-form")
+    private Div passwordForm;
+
+    @Inject
+    @DataField("password")
+    private TextInput password;
+
+    @Inject
+    @DataField("test-connection-button")
+    private Button testConnectionButton;
+
+    @Inject
+    private TranslationService translationService;
+
+    @Inject
+    private PopupHelper popupHelper;
+
+    private WF10ProviderConfigPresenter presenter;
+
+    @PostConstruct
+    private void init() {
+        providerTypeName.setTextContent(getWizardTitle());
+    }
+
+    @Override
+    public void init(final WF10ProviderConfigPresenter presenter) {
+        this.presenter = presenter;
+    }
+
+    @Override
+    public String getProviderName() {
+        return name.getValue();
+    }
+
+    @Override
+    public String getHost() {
+        return host.getValue();
+    }
+
+    @Override
+    public String getPort() {
+        return port.getValue();
```

Comment:
Playing with the feature some more, I think it would be useful to provide pre-filled default values for the ports. Most of the time the management port will be 9990 and port will be 8080 and user might change them if he needs to. @wmedvede Wdyt?